### PR TITLE
[Keyboard] add Soup10 support

### PR DESCRIPTION
**Pull request by Sadeeed**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/soup10/config.h [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x5375
+#define PRODUCT_ID   0x0001
+#define DEVICE_VER   0x0001
+#define MANUFACTURER icesoup
+#define PRODUCT      soup10
+#define DESCRIPTION  Soup10
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 1
```
Comment (zvecr) [inline keyboards/soup10/config.h:30]:
```suggestion
```

File: keyboards/soup10/config.h [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
+/* Bootmagic Lite key configuration */
+//#define BOOTMAGIC_LITE_ROW 0
+//#define BOOTMAGIC_LITE_COLUMN 0
```
Comment (zvecr) [inline keyboards/soup10/config.h:156]:
```suggestion
#define BOOTMAGIC_LITE_ROW 0
#define BOOTMAGIC_LITE_COLUMN 1
```

File: keyboards/soup10/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+//             if (record->event.pressed) {
+//                 // when keycode QMKURL is pressed
+//                 SEND_STRING("https://qmk.fm/\n");
+//             } else {
+//                 // when keycode QMKURL is released
+//             }
+//             break;
+//     }
+//     return true;
+// }
```
Comment (zvecr) [inline keyboards/soup10/keymaps/default/keymap.c:65]:
```suggestion
```

File: keyboards/soup10/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    _BASE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+// enum custom_keycodes {
+//     QMKBEST = SAFE_RANGE,
+//     QMKURL
+// };
+
```
Comment (zvecr) [inline keyboards/soup10/keymaps/default/keymap.c:29]:
```suggestion
```

File: keyboards/soup10/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        KC_A, 
+        KC_B, KC_C, KC_D, 
+        KC_E, KC_F, KC_G, 
+        KC_H, KC_I, KC_J),
+
+    [_FN] = LAYOUT(
```
Comment (zvecr) [inline keyboards/soup10/keymaps/default/keymap.c:38]: Theres no way to actually get to this layer, should either be removed, or a way added from _BASE

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/soup10/readme.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# soup10
+
+![soup10](https://i.imgur.com/YIMB3kK.jpg)
```
Comment (fauxpark) [inline keyboards/soup10/readme.md:3]:
```suggestion
![soup10](https://i.imgur.com/YIMB3kKl.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]
LGTM

**Comment by drashna** [comment]
Thank you for your contribution!

**Pull request merged**